js/frontend.js

```diff
@@ -1,41 +1,154 @@
+const form = document.querySelector("form");
+const formBookTitle = document.querySelector("#bookTitle");
+const formWordGoals = document.querySelector("#wordGoals");
+const formDeadline = document.querySelector("#deadline");
+const booksDiv = document.querySelector("#books");
+const bookSubmissions = document.querySelector("#book-submissions");
 
+const serverUrl = "http://localhost:3000/addbook";
 
-const form = document.querySelector('form')
-const serverUrl = 'http://localhost:3000/addbook'
+form.addEventListener("submit", e => {
+  e.preventDefault();
 
+  const formData = new FormData(form);
+  const bookTitle = formData.get("bookTitle");
+  const wordGoals = formData.get("wordGoals");
+  const deadline = formData.get("deadline");
 
-form.addEventListener('submit', (e) => {
-    e.preventDefault()
-    const formData = new FormData(form)    
-    const bookTitle = formData.get('bookTitle')
-    const wordGoals = formData.get('wordGoals')
-    const deadline = formData.get('deadline')
-    
-    const bookEntry = {
-        bookTitle,
-        wordGoals,
-        deadline
+  if (formBookTitle.value === "" || formWordGoals.value === "") {
+    alert("Fields are required.");
+  }
+
+  const bookEntry = {
+    bookTitle,
+    wordGoals,
+    deadline: new Date(deadline)
+  };
+
+  async function postData(url, data = {}) {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    return response.json();
+  }
+
+  postData(serverUrl, bookEntry)
+    .then(data => console.log(data))
+    .catch(error => console.error(error));
+
+  console.log(bookEntry);
+  formBookTitle.value = "";
+  formWordGoals.value = "";
+  document.querySelector("#books").innerHTML = "";
+  setTimeout(() => {
+    getBooks(serverUrl);
+  }, 0300);
+});
+
+async function getBooks(url) {
+  try {
+    const response = await fetch(url, {
+      method: "GET",
```

Comment:
For a simple get request, no need for the options - method or header. Can just pass in the url.